### PR TITLE
FIX: Fixes for Py35

### LINKED ISSUE #2603: [py3] Brainvision tests fails due UnicodeDecodeError
I'm using last released version of MNE (probably you will thnk that it's because I unbundled libs -> no. this build with original release).
```python
======================================================================
ERROR: Test reading raw Brain Vision files
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/test_brainvision.py", line 35, in test_brainvision_data_filters
    preload=True)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 511, in read_raw_brainvision
    response_trig_shift=response_trig_shift)
  File "<string>", line 2, in __init__
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/utils.py", line 549, in verbose
    return function(*args, **kwargs)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 79, in __init__
    vhdr_fname, eog, misc, response_trig_shift, scale)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 303, in _get_vhdr_info
    l = f.readline().strip()
  File "/usr/lib64/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
nose.proxy.UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 638: ordinal not in range(128)
-------------------- >> begin captured stdout << ---------------------
Extracting parameters from /builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/data/test_highpass.vhdr...
--------------------- >> end captured stdout << ----------------------
======================================================================
ERROR: Test reading raw Brain Vision files
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/test_brainvision.py", line 48, in test_brainvision_data
    raw_py = read_raw_brainvision(vhdr_path, montage, eog=eog, preload=True)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 511, in read_raw_brainvision
    response_trig_shift=response_trig_shift)
  File "<string>", line 2, in __init__
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/utils.py", line 549, in verbose
    return function(*args, **kwargs)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 79, in __init__
    vhdr_fname, eog, misc, response_trig_shift, scale)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 303, in _get_vhdr_info
    l = f.readline().strip()
  File "/usr/lib64/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
nose.proxy.UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 638: ordinal not in range(128)
-------------------- >> begin captured stdout << ---------------------
Extracting parameters from /builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/data/test.vmrk...
Extracting parameters from /builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/data/test.vhdr...
Extracting parameters from /builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/data/test.vhdr...
--------------------- >> end captured stdout << ----------------------
======================================================================
ERROR: Test reading and modifying events
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/test_brainvision.py", line 92, in test_events
    raw = read_raw_brainvision(vhdr_path, eog=eog, preload=True)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 511, in read_raw_brainvision
    response_trig_shift=response_trig_shift)
  File "<string>", line 2, in __init__
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/utils.py", line 549, in verbose
    return function(*args, **kwargs)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 79, in __init__
    vhdr_fname, eog, misc, response_trig_shift, scale)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 303, in _get_vhdr_info
    l = f.readline().strip()
  File "/usr/lib64/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
nose.proxy.UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 638: ordinal not in range(128)
-------------------- >> begin captured stdout << ---------------------
Extracting parameters from /builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/data/test.vhdr...
--------------------- >> end captured stdout << ----------------------
======================================================================
ERROR: Test writing raw eeg files when preload is False
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.4/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/test_brainvision.py", line 179, in test_read_segment
    raw1 = read_raw_brainvision(vhdr_path, eog=eog, preload=False)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 511, in read_raw_brainvision
    response_trig_shift=response_trig_shift)
  File "<string>", line 2, in __init__
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/utils.py", line 549, in verbose
    return function(*args, **kwargs)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 79, in __init__
    vhdr_fname, eog, misc, response_trig_shift, scale)
  File "/builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/brainvision.py", line 303, in _get_vhdr_info
    l = f.readline().strip()
  File "/usr/lib64/python3.4/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
nose.proxy.UnicodeDecodeError: 'ascii' codec can't decode byte 0xc2 in position 638: ordinal not in range(128)
-------------------- >> begin captured stdout << ---------------------
Extracting parameters from /builddir/build/BUILD/python3-python-mne-0.10-1.fc24/mne/io/brainvision/tests/data/test.vhdr...
--------------------- >> end captured stdout << ----------------------
----------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #2603.

@ignatenkobrain can you see if this fixes your issue?

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -1252,14 +1258,14 @@ def next(self, return_event_id=False):
         """
         if self.preload:
             if self._current >= len(self._data):
-                raise StopIteration
```
Comment (Eric89GXL) [inline mne/epochs.py]: Apparently using `StopIteration` is a ["hack"] (https://www.python.org/dev/peps/pep-0479/) and we needed to refactor these functions, so I've done that here.

File: mne/viz/epochs.py [hunk 2]
```diff
@@ -909,7 +909,7 @@ def _plot_update_epochs_proj(params, bools=None):
     params['projector'], _ = setup_proj(params['info'], add_eeg_ref=False,
                                         verbose=False)
 
-    start = params['t_start'] / len(params['epochs'].times)
+    start = int(params['t_start'] / len(params['epochs'].times))
```
Comment (Eric89GXL) [inline mne/viz/epochs.py]: @jaeilepp can you verify this was correct? It's used to create a slice into `Epochs` so I think it's supposed to be an integer, but I'm not sure whether you intended truncation vs rounding here. (I was getting warnings about slices having non-integer values.)

File: .travis.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ env:
     # but breaks scipy; hopefully eventually the NUMPY=1.9 on 2.7 full can be removed
     - PYTHON=2.7 DEPS=full TEST_LOCATION=src NUMPY="=1.9" SCIPY="=0.16"
     - PYTHON=2.7 DEPS=nodata TEST_LOCATION=src MNE_DONTWRITE_HOME=true  # also runs flake8
-    - PYTHON=3.4 DEPS=full TEST_LOCATION=install MNE_STIM_CHANNEL=STI101
+    - PYTHON=3.5 DEPS=full TEST_LOCATION=install MNE_STIM_CHANNEL=STI101
```
Comment (ignatenkobrain) [inline .travis.yml]: why not both?

Comment (Eric89GXL) [inline .travis.yml]: increases build time, which is already a bit long for us

**Comment by Eric89GXL** [comment]
Travis passed but the 3.5 build took an hour 15 minutes, I'm going to restart it to see if it was a fluke.

**Comment by agramfort** [comment]
LGTM

+1 for merge

if we can make travis as fast as with 3.4

we'll need to backport for package.

we'll make a 0.10.1 release.

**Comment by Eric89GXL** [comment]
Yeah I realized the slowness came from accidentally having coverage on for the 3.5 build (since the conditional looked for "3.4"), re-running now. I'll squash and merge if the time is reasonable, then push the commit to maint/0.10.

**Comment by agramfort** [comment]
sounds good thx

**Comment by Eric89GXL** [comment]
merged and backported

File: mne/viz/epochs.py (hunk 2 above)
Comment (jaeilepp) [inline mne/viz/epochs.py]: Yeah, I think you got it right.

**Pull request merged**